feat: add `strided` and `assign` APIs to `complex/float32/base/add`

ShabiShett07 opened the pull request:
Resolves #5196 

## Description

> What is the purpose of this pull request?

This pull request:

-   Solves the issue #5196 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5196 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
feat: add /complex/float32/base/add

ShabiShett07 committed:
feat: add /complex/float32/base/add

ShabiShett07 committed:
docs: files completed

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

ShabiShett07 committed:
fix: update test.strided.js

ShabiShett07 committed:
fix: update test.strided.js

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
@Neerajpathak07 could you help me out with the CI errors?

Neerajpathak07 commented:
```
Error: Cannot find module '@stdlib/assert/is-same-Float32Array'
```
- For starters checkout if this latter exists in the repo or not.

- For Lint errors most of them are based on needing a newline at the end of the file. The details should provide a brief overview to it.

Neerajpathak07 commented on lib/node_modules/@stdlib/complex/float32/base/add/test/test.strided.js:
```diff
@@ -0,0 +1,157 @@
+	expected = new Float32Array( [ 3.0, 4.0 ] );
+
+	t.deepEqual( v, out, 'returns expected value' );
+	t.deepEqual( isSameFloat32Array( out, expected ), true, 'returns expected value' );
```
It takes a toll on the tests for all of the tests which have `isSameFloat32Array()` function as the input. So resolving this should make the tests run clean.

Neerajpathak07 commented:
@ShabiShett07 ahh understood where the problem should lie so the latter does exist but it appears you have made a typo the correct module should be :- `@stdlib/assert/is-same-float32array`
This should resolve the error. LMK if it still persists.

ShabiShett07 committed:
fix: update test.assign.js

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

Neerajpathak07 commented on lib/node_modules/@stdlib/complex/float32/base/add/test/test.assign.js:
```diff
@@ -0,0 +1,132 @@
+// MODULES //
+
+var tape = require( 'tape' );
+var isSameFloat32Array = require( '@stdlib/assert/is-same-lfoat32array' );
```
```suggestion
var isSameFloat32Array = require( '@stdlib/assert/is-same-foat32array' );
```
also make sure you have made the changes for every file in which this module was used.

ShabiShett07 replied:
> also make sure you have made the changes for every file in which this module was used.

I have fixed this, but still it is failing in test.strided.js

Neerajpathak07 replied:
Yess, the error related to module has been resolved now there is something more to it.

ShabiShett07 committed:
fix: update test.assign.js

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

Neerajpathak07 commented on lib/node_modules/@stdlib/complex/float32/base/add/test/test.strided.js:
```diff
@@ -0,0 +1,157 @@
+
+var tape = require( 'tape' );
+var isSameFloat32Array = require( '@stdlib/assert/is-same-float32array' );
+var Float32Array = require( '@stdlib/array/float64' );
```
```suggestion
var Float32Array = require( '@stdlib/array/float32' );
```
you were meant to have a `Float32Array` but it was being called form the incorrect package. Due to which the `expected` var, ended up having a double-precision value.

ShabiShett07 committed:
fix: update test.strided.js

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
Thank you @Neerajpathak07, this solved my error.

ShabiShett07 commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

kgryte commented on lib/node_modules/@stdlib/complex/float32/base/add/README.md:
```diff
@@ -206,14 +261,6 @@ int main( void ) {
--   <span class="package-name">[`@stdlib/complex/float32/base/mul`][@stdlib/complex/float32/base/mul]</span><span class="delimiter">: </span><span class="description">multiply two single-precision complex floating-point numbers.</span>
--   <span class="package-name">[`@stdlib/math/base/ops/csubf`][@stdlib/math/base/ops/csubf]</span><span class="delimiter">: </span><span class="description">subtract two single-precision complex floating-point numbers.</span>
-
 </section>
```
```suggestion
* * *

## See Also

-   <span class="package-name">[`@stdlib/complex/float64/base/add`][@stdlib/complex/float64/base/add]</span><span class="delimiter">: </span><span class="description">add two double-precision complex floating-point numbers.</span>
-   <span class="package-name">[`@stdlib/complex/float32/base/mul`][@stdlib/complex/float32/base/mul]</span><span class="delimiter">: </span><span class="description">multiply two single-precision complex floating-point numbers.</span>
-   <span class="package-name">[`@stdlib/math/base/ops/csubf`][@stdlib/math/base/ops/csubf]</span><span class="delimiter">: </span><span class="description">subtract two single-precision complex floating-point numbers.</span>

</section>
```

This shouldn't have been removed.

kgryte commented on lib/node_modules/@stdlib/complex/float32/base/add/README.md:
```diff
@@ -224,12 +271,6 @@ int main( void ) {
-
-[@stdlib/math/base/ops/csubf]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/ops/csubf
-
 <!-- </related-links> -->
```
```suggestion
[@stdlib/complex/float64/base/add]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/complex/float64/base/add

[@stdlib/complex/float32/base/mul]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/complex/float32/base/mul

[@stdlib/math/base/ops/csubf]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/ops/csubf

<!-- </related-links> -->
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/complex/float32/base/add/benchmark/benchmark.asssign.js:
This file is incorrectly named. s/asssign/assign/

kgryte commented on lib/node_modules/@stdlib/complex/float32/base/add/benchmark/benchmark.js:
Why were the permissions on this file changed?

kgryte commented on lib/node_modules/@stdlib/complex/float32/base/add/benchmark/benchmark.native.js:
Why were the permissions on this file changed?

kgryte replied:
Same thing for all these files. Please revert all the permissions changes. This PR otherwise cannot move forward.

ShabiShett07 committed:
fix: rename benchmark.assign.js

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: updated file permissions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
Sorry @kgryte sir for such inconvenience, I have made the necessary changes. Please check

ShabiShett07 committed:
fix: updated file permissions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: failed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: passed
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: passed
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

ShabiShett07 committed:
fix: updated file permissions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

ShabiShett07 commented:
Sorry @kgryte, I think this PR still needs some changes I will notify once it is done

ShabiShett07 committed:
Merge branch 'feature/float32/add' of https://github.com/ShabiShett07/stdlib into feature/float32/add

ShabiShett07 committed:
fix: changed permissions

ShabiShett07 committed:
fix: changed permissions

ShabiShett07 committed:
fix: changed permissions

ShabiShett07 commented:
@kgryte, Closed this issue because it has lot of errors and opened a new one #5391 which is ready for review
